d2/src/main/java/com/linkedin/d2/xds/GlobCollectionUtils.java

```diff
@@ -1,41 +1,77 @@
 package com.linkedin.d2.xds;
 
+import javax.annotation.Nullable;
+
 public class GlobCollectionUtils
 {
-  private static final String D2_URI_NODE_GLOB_COLLECTION_PREFIX = "xdstp:///indis.D2URI/";
+  private static final String D2_URIS_PREFIX = "/d2/uris/";
+  private static final String D2_URI_NODE_GLOB_COLLECTION_PREFIX = "xdstp:///indis.D2URI/d2/uris/";
   private static final String GLOB_COLLECTION_SUFFIX = "/*";
 
   private GlobCollectionUtils()
   {
   }
 
-  /**
-   * Extracts the glob collection URL (e.g. xdstp:///indis.D2URI/FooCluster/*) from the given resource URN.
-   */
-  public static String globCollectionUrlFromResourceUrn(String resourceUrn)
+  public static class D2UriIdentifier
   {
-    return resourceUrn.substring(0, resourceUrn.lastIndexOf('/')) + GLOB_COLLECTION_SUFFIX;
-  }
+    private final String _clusterPath;
```

Comment:
please add some comment for the expected format of this.

Reply:
Done